### ISSUE TITLE
Keeps reconnecting on iPad

### ISSUE THREAD

**Issue by korjwl1** [coder/code-server#3580]
## OS/Web Information

- Web Browser: Safari, Edge, . .. etc
- Local OS: Lubuntu 20.04
- Remote OS: IpadOS 15 on M1 Ipad
- Remote Architecture: I used code-server with bind-addr option and cert, cert-key option. Used pem files from Let's Encrypt for my asus router DDNS for cert and cert-key
- `code-server --version`: 3.10.2

## Steps to Reproduce

1. Just access with any browsers on IpadOS 15
2. Then I cannot access to any files, extensions and terminal on the  Code-Server. All I can see is keep reconnecting response on the left bottom part. (The one for open remote repository)
3.

## Expected

 The code-server works perfectly on the Microsoft Edge on Windows 10 and Firefox on Lubuntu 20.04.
I also expected that on the Safari or Microsoft Edge of Ipad OS 15

## Actual

 I didn't see the white page, and could have clicked the ui buttons. However, when I click new terminal, the terminal partition comes out but no terminal is opened. Also, I cannot access to the folders, files and also cannot see any extensions. 
All I could see was that on the left bottom part where you can see your domain, the status message kept changing between reconnecting to your doamin and your domain rapidly.

## Logs

 6월 09 23:34:21 lubuntu code-server[1561]:     at process.emit (events.js:314:20)
 6월 09 23:34:21 lubuntu code-server[1561]: Error: Unexpected SIGPIPE
 6월 09 23:34:21 lubuntu code-server[1561]:     at process.<anonymous> (/usr/lib/code-server/lib/vscode/out/bootstrap.js:1:370)
 6월 09 23:34:21 lubuntu code-server[1561]:     at process.emit (events.js:314:20)
 6월 09 23:34:21 lubuntu code-server[1561]: [2021-06-09T14:34:21.301Z] warn  vscode Uncaught error {"error":"read ECONNRESET"}
 6월 09 23:34:21 lubuntu code-server[1561]: Error: Unexpected SIGPIPE
 6월 09 23:34:21 lubuntu code-server[1561]:     at process.<anonymous> (/usr/lib/code-server/lib/vscode/out/bootstrap.js:1:370)
 6월 09 23:34:21 lubuntu code-server[1561]:     at process.emit (events.js:314:20)
 6월 09 23:34:24 lubuntu code-server[1561]:   ERR [File Watcher (chokidar)] Inotify limit reached (ENOSPC)
 6월 09 23:34:24 lubuntu code-server[1561]:   ERR [File Watcher (chokidar)] Inotify limit reached (ENOSPC)

Above is the log from the systemctl

## Notes

This issue can be reproduced in VS Code: Yes/No

**Comment by jsjoeio** [comment]
Hmm... I'll need to take a look at my own iPad setup and see if I'm having similar issues.

Is it possible for you to take a screen recording on your iPad and share here? That might help us troubleshoot.

**Comment by jmlingeman** [comment]
Getting the same error here on iOS 15/MacOS 12 betas in Safari. Specifically seems to happen when interacting with the extensions search bar.

`code-server | [2021-06-09T21:03:52.296Z] info  code-server 3.10.2 387b12ef4ca404ffd39d84834e1f0776e9e3c005
code-server | [2021-06-09T21:03:52.297Z] info  Using user-data-dir ~/.local/share/code-server
code-server | [2021-06-09T21:03:52.305Z] info  Using config file ~/.config/code-server/config.yaml
code-server | [2021-06-09T21:03:52.305Z] info  HTTP server listening on http://0.0.0.0:8080 
code-server | [2021-06-09T21:03:52.305Z] info    - Authentication is disabled 
code-server | [2021-06-09T21:03:52.306Z] info    - Not serving HTTPS 
code-server | [2021-06-09T21:03:52.602Z] error vscode is not running Error: vscode is not running
code-server |     at VscodeProvider.send (/usr/lib/code-server/out/node/vscode.js:206:19)
code-server |     at VscodeProvider.<anonymous> (/usr/lib/code-server/out/node/vscode.js:198:30)
code-server |     at step (/usr/lib/code-server/out/node/vscode.js:63:23)
code-server |     at Object.next (/usr/lib/code-server/out/node/vscode.js:44:53)
code-server |     at fulfilled (/usr/lib/code-server/out/node/vscode.js:35:58)
code-server |     at processTicksAndRejections (internal/process/task_queues.js:97:5)
code-server | [2021-06-09T21:03:52.621Z] error vscode is not running Error: vscode is not running
code-server |     at VscodeProvider.send (/usr/lib/code-server/out/node/vscode.js:206:19)
code-server |     at VscodeProvider.<anonymous> (/usr/lib/code-server/out/node/vscode.js:198:30)
code-server |     at step (/usr/lib/code-server/out/node/vscode.js:63:23)
code-server |     at Object.next (/usr/lib/code-server/out/node/vscode.js:44:53)
code-server |     at fulfilled (/usr/lib/code-server/out/node/vscode.js:35:58)
code-server |     at processTicksAndRejections (internal/process/task_queues.js:97:5)
code-server | Error: Unexpected SIGPIPE
code-server |     at process.<anonymous> (/usr/lib/code-server/lib/vscode/out/bootstrap.js:1:370)
code-server |     at process.emit (events.js:314:20)
code-server | Error: Unexpected SIGPIPE
code-server |     at process.<anonymous> (/usr/lib/code-server/lib/vscode/out/bootstrap.js:1:370)
code-server |     at process.emit (events.js:314:20)
code-server | Error: Unexpected SIGPIPE
code-server |     at process.<anonymous> (/usr/lib/code-server/lib/vscode/out/bootstrap.js:1:370)
code-server |     at process.emit (events.js:314:20)
code-server | Error: Unexpected SIGPIPE
code-server |     at process.<anonymous> (/usr/lib/code-server/lib/vscode/out/bootstrap.js:1:370)
code-server |     at process.emit (events.js:314:20)
code-server | Error: Unexpected SIGPIPE
code-server |     at process.<anonymous> (/usr/lib/code-server/lib/vscode/out/bootstrap.js:1:370)
code-server |     at process.emit (events.js:314:20) (this just keeps repeating forever)`

**Comment by jsjoeio** [comment]
> Getting the same error here on iPadOS 15/MacOS 12 betas in Safari

Hmm... I wonder if something changed in the newest versions of Safari 🤔

I just tested 3.10.2 on my iPad today (14.6) and it works as expected. I'm also using it on macOS Big Sur 11.3 and having no

## Screenshots

https://user-images.githubusercontent.com/3806031/121418548-e9495b00-c91f-11eb-9a3c-03037802e3e4.MP4

![image](https://user-images.githubusercontent.com/3806031/121432721-e0608580-c92f-11eb-929e-e6e6b74b02c7.png)

**Comment by jmlingeman** [comment]
Yeah it looks like it is likely an updated Safari problem. At least in my case if I do not try to search for any extensions then everything seems to work fine. Searching for extensions causes the above behavior. Not sure if that is the case for OP.

**Comment by jsjoeio** [comment]
> Searching for extensions causes the above behavior

Very strange. Maybe @code-asher has seen something like this in the past

**Comment by korjwl1** [comment]
![ezgif com-video-to-gif](https://user-images.githubusercontent.com/72062804/121453482-368b0280-c9dc-11eb-9863-2bd5dbe0ca5d.gif)

this is the recorded screen. I took more time to investigate, and found out that such behavior only happens after I try to search for the extensions as other guy said. The problem is that, after such thing happens, only after I shut down the server the code-server comes back to normal on Ipad

> Hmm... I'll need to take a look at my own iPad setup and see if I'm having similar issues.
> 
> Is it possible for you to take a screen recording on your iPad and share here? That might help us troubleshoot.

**Comment by reticulatingsplines** [comment]
Having similar issues on my iPad as well. Running iOS 14.6 on an iPad Pro 11” 2018 model, code-server 3.10.2 being hosted on arm64 Debian buster and installed via install.sh. This only happens when I try to access remotely over the internet…if I VPN into the LAN that my server is located on and access code-server via the local address, the problem disappears. Perhaps a weird new SSL issue?

**Comment by jsjoeio** [comment]
Thank you all for the screen recordings!

This is very very strange... But the fact that multiple of you are having the same issues means something has to be up 🤔

> such behavior only happens after I try to search for the extensions

I just tested with my iPad setup (iPad iOS 14.6 + code-server 3.10.2 running on Railway) and it works as expected

https://user-images.githubusercontent.com/3806031/121562289-29671700-c9ce-11eb-942d-1a4e9c426deb.MP4

@code-asher or @oxy any ideas on what might be causing this? The only thing coming to my mind is issues with native dependencies i.e. `ripgrep` but I think that only affects code-search, not extension-search 🤔

The only other thing I'm noticing in common between @korjwl1 and @reticulatingsplines videos is in the bottom left corner - see how it keeps flashing saying "Reconnecting"? Notice how mine never changes. My guess is it has to do somehow with the connection to the environment? But unsure about that. 

Have any of you [reconfigured the marketplace URL](https://github.com/cdr/code-server/blob/main/docs/FAQ.md#how-do-i-configure-the-marketplace-url)?

**Comment by jmlingeman** [comment]
Here is the error that appears in the JS console (using the Inspect Browser app). 
![Screenshot 2021-06-10 at 6 41 11 PM](https://user-images.githubusercontent.com/522069/121606210-bbe1d780-ca1b-11eb-838a-244fc64f5c98.png)

I’m not sure how to enable the trace logs mentioned in the image, but I would be happy to do so. I also can’t seem to find where the code-server logs are in the coder provided docker container. Again, would be happy to grab those if someone can point me to where they are.

**Comment by korjwl1** [comment]
> The only other thing I'm noticing in common between @korjwl1 and @reticulatingsplines videos is in the bottom left corner - see how it keeps flashing saying "Reconnecting"? Notice how mine never changes. My guess is it has to do somehow with the connection to the environment? But unsure about that.
> 
> Have any of you [reconfigured the marketplace URL](https://github.com/cdr/code-server/blob/main/docs/FAQ.md#how-do-i-configure-the-marketplace-url)?

Here's my environment
Code-server
- OS: Lubuntu 20.04 (Basically same as the Ubuntu 20.04) on Synology VMM
- Code-Server Version: 3.10.2
- Installation Method: curl -fsSL https://code-server.dev/install.sh | sh
- Settings: used options of '--bind-addr', 'cert', 'cert-key'. Let's Encrypt certificate was used. Nothing else used

Tested Environments
- Windows 10: Worked without any problem with Microsoft Edge
- macOS Big Sur 11.4: Worked without any problem with Safari, Vivaldi
- Android: Worked without any problem with Kiwi Browser
- IpadOS 15.0 Beta with iPad 12.9 2021: Reconnecting issue on any browsers even on the local network
- Lubuntu20.04: 


  The problem must be caused by the extension market since I can reproduce the problem by opening the extension market. The problem can be solved by connecting to the code-server on the other environments and open any files so that the last visited page is not the extension market. If you just move to other tab and do not open any files, the problem won't be solved.
 I do not know what part of IpadOS 15.0 caused the unexpected SIGPIPE but this seems to be the reason. I'll borrow my friends iPad with iPadOS 14 and then post the result on here.

**Comment by jmlingeman** [comment]
Since this also happens on MacOS 12.0 Beta in Safari (by loading the extensions tab), here are the logs from there, which match the error logs from the iPad:

The error messages on MacOS Safari begin with this error message:

<img width="782" alt="Screen Shot 2021-06-11 at 10 30 17" src="https://user-images.githubusercontent.com/522069/121702664-16247c00-caa0-11eb-90fe-6944d222c0ef.png">

And then these messages repeat:

```[Error] WebSocket connection to 'ws://192.168.1.4:8443/?type=Management&reconnectionToken=1ca5b7bf-701e-41b4-993d-bcc8add38d8f&reconnection=true&skipWebSocketFrames=false' failed: The operation couldn’t be completed. (kNWErrorDomainPOSIX error 40 - Message too long)
[Log]  INFO – "[remote-connection][Management   ][1ca5b…][reconnect] starting reconnecting loop. You can get more information with the trace log level." (workbench.web.api.js, line 627)
[Log]  INFO – "[remote-connection][Management   ][1ca5b…][reconnect] resolving connection..." (workbench.web.api.js, line 627)
[Log]  INFO – "[remote-connection][Management   ][1ca5b…][reconnect] connecting to 192.168.1.4:8443..." (workbench.web.api.js, line 627)
[Log]  INFO – "[remote-connection][Management   ][1ca5b…][reconnect] reconnected!" (workbench.web.api.js, line 627)```

**Comment by jsjoeio** [comment]
@korjwl1 thank you for the thorough testing and the notes!

@jmlingeman thank you for the logs and the screenshots! And the fact that you can reproduce on macOS + Safari is great because that means we can test easily (I'm on macOS). 

cc @code-asher - I know you've fixed Safari issues in the past. Maybe you see something obvious here 👀

**Comment by code-asher** [comment]
I've never seen that "message too long" error before so I suppose that
could be related. I wonder if the beta has a bug or if it has much lower
limits than it previously did.

Lists of extensions can be quite large so that could explain why it only
happens when viewing the extensions sidebar.

**Comment by oxkenshin** [comment]
In my case, it has something to do with the extension panel(tab). It works fine when the app starts with the file explorer panel, but failed to connect when I switch to extension panel. Sadly I don't know how to remove the installed extensions without switching to extension panel.

**Comment by Cattttttttt** [comment]
I got the same error when I was using safari on iPadOS 15 beta.
Then I changed to serveditor but this issue continues, and it happened in Firefox and Chrome for iPadOS 15
Maybe it's a bug of new webkit?

Firefox on windows worked without any problems

code-server is running on Debian 10 on WSL2

```
[2021-06-15T14:50:50.075Z] info  code-server 3.10.2
[2021-06-15T14:50:50.076Z] info  Using user-data-dir ~/.local/share/code-server
[2021-06-15T14:50:50.084Z] info  Using config file ~/.config/code-server/config.yaml
[2021-06-15T14:50:50.084Z] info  HTTP server listening on http://0.0.0.0:8080
[2021-06-15T14:50:50.084Z] info    - Authentication is disabled
[2021-06-15T14:50:50.084Z] info    - Not serving HTTPS
Error: Unexpected SIGPIPE
    at process.<anonymous> (/usr/lib/code-server/lib/vscode/out/bootstrap.js:1:370)
    at process.emit (events.js:314:20)
Error: Unexpected SIGPIPE
    at process.<anonymous> (/usr/lib/code-server/lib/vscode/out/bootstrap.js:1:370)
    at process.emit (events.js:314:20)
Error: Unexpected SIGPIPE
    at process.<anonymous> (/usr/lib/code-server/lib/vscode/out/bootstrap.js:1:370)
    at process.emit (events.js:314:20)
Error: Unexpected SIGPIPE
    at process.<anonymous> (/usr/lib/code-server/lib/vscode/out/bootstrap.js:1:370)
    at process.emit (events.js:314:20)
Error: Unexpected SIGPIPE
    at process.<anonymous> (/usr/lib/code-server/lib/vscode/out/bootstrap.js:1:370)
    at process.emit (events.js:314:20)
Error: Unexpected SIGPIPE
    at process.<anonymous> (/usr/lib/code-server/lib/vscode/out/bootstrap.js:1:370)
    at process.emit (events.js:314:20)
Error: Unexpected SIGPIPE
    at process.<anonymous> (/usr/lib/code-server/lib/vscode/out/bootstrap.js:1:370)
    at process.emit (events.js:314:20)
[2021-06-15T14:58:11.699Z] warn  vscode Uncaught error {"error":"read ECONNRESET"}
[2021-06-15T14:58:19.072Z] warn  vscode Uncaught error {"error":"read ECONNRESET"}
Error: Unexpected SIGPIPE
    at process.<anonymous> (/usr/lib/code-server/lib/vscode/out/bootstrap.js:1:370)
    at process.emit (events.js:314:20)
Error: Unexpected SIGPIPE
    at process.<anonymous> (/usr/lib/code-server/lib/vscode/out/bootstrap.js:1:370)
    at process.emit (events.js:314:20)
Error: Unexpected SIGPIPE
    at process.<anonymous> (/usr/lib/code-server/lib/vscode/out/bootstrap.js:1:370)
    at process.emit (events.js:314:20)
Error: Unexpected SIGPIPE
    at process.<anonymous> (/usr/lib/code-server/lib/vscode/out/bootstrap.js:1:370)
    at process.emit (events.js:314:20)
[2021-06-15T14:58:54.546Z] warn  vscode Uncaught error {"error":"read ECONNRESET"}
```

**Comment by jsjoeio** [comment]
> Maybe it's a bug of new webkit

Yeah I think you might be right. I'm still on 14.5 on my iPad and not seeing these issues

**Comment by idoodler** [comment]
I am also running iOS 15 on my new iPad 2021 and I am not experiencing such issues. You might try out the newest [Safari Technology Preview](https://developer.apple.com/safari/technology-preview/) build. It also uses the new WebKit. In addition you can try and open the Development Tools, you may see an interresting log message.

I am running Codeserver on an vServer running Ubuntu.

**Comment by MrRorg** [comment]
I have the same issue. iPad Air 4 2020. iPadOS 15 b1. Continuously receiving the message in server console:
Error: Unexpected SIGPIPE
    at process.<anonymous> (/usr/lib/code-server/lib/vscode/out/bootstrap.js:1:370)
    at process.emit (events.js:314:20)

Server located on DigitalOcean. Ubuntu 21.04. Connected remotely. Start code-server using code-server --bind-addr <ip>:8080

**Comment by reticulatingsplines** [comment]
Here's what I see when I run code-server as a process (vs daemon) with the verbose option, then try to connect from my iPad Pro running iOS 14.6 (final line SIGINT was from me intentionally ctrl-c'ing, fyi):

pi@raspberrypi:~ $ code-server --verbose
[2021-06-25T03:52:24.699Z] trace child:32529 got message {"message":{"type":"handshake","args":{"_":[],"bind-addr":"127.0.0.1:8080","auth":"password","password":"REDACTED","config":"/home/pi/.config/code-server/config.yaml","verbose":true,"user-data-dir":"/home/pi/.local/share/code-server","extensions-dir":"/home/pi/.local/share/code-server/extensions","log":"trace","host":"127.0.0.1","port":8080,"proxy-domain":[],"usingEnvPassword":false,"usingEnvHashedPassword":false}}} 
[2021-06-25T03:52:24.708Z] info  code-server 3.10.2 387b12ef4ca404ffd39d84834e1f0776e9e3c005                            
[2021-06-25T03:52:24.709Z] info  Using user-data-dir ~/.local/share/code-server                                         
[2021-06-25T03:52:24.710Z] trace Using extensions-dir ~/.local/share/code-server/extensions                             
[2021-06-25T03:52:24.749Z] info  Using config file ~/.config/code-server/config.yaml                                    
[2021-06-25T03:52:24.749Z] info  HTTP server listening on http://127.0.0.1:8080                                         
[2021-06-25T03:52:24.750Z] info    - Authentication is enabled                                                          
[2021-06-25T03:52:24.750Z] info      - Using password from ~/.config/code-server/config.yaml                            
[2021-06-25T03:52:24.750Z] info    - Not serving HTTPS                                                                  
[2021-06-25T03:52:29.969Z] trace heartbeat                                                                              
[2021-06-25T03:52:29.987Z] debug redirecting from / to ./login                                                          
[2021-06-25T03:52:36.430Z] debug got cookie doman {"host":"vbc.shmoo.dev"}                                              
[2021-06-25T03:52:36.433Z] debug redirecting from /login to ./                                                          
[2021-06-25T03:52:36.597Z] debug forking vs code...                                                                     
[2021-06-25T03:52:37.510Z] trace got message {"message":{"type":"ready"}}                                               
[2021-06-25T03:52:37.511Z] debug setting up vs code...                                                                  
[2021-06-25T03:52:37.515Z] debug vscode got message from code-server {"type":"init"}                                    
[2021-06-25T03:52:37.519Z] trace vscode code-server message content {"message":{"type":"init","id":"PVkS1Xb5TRcuzXLB4FI1QbvO","options":{"args":{"_":[],"bind-addr":"127.0.0.1:8080","auth":"password","password":REDACTED,"config":"/home/pi/.config/code-server/config.yaml","verbose":true,"user-data-dir":"/home/pi/.local/share/code-server","extensions-dir":"/home/pi/.local/share/code-server/extensions","log":"trace","host":"127.0.0.1","port":8080,"proxy-domain":[],"usingEnvPassword":false,"usingEnvHashedPassword":false},"remoteAuthority":"vbc.shmoo.dev","startPath":{"url":"/media/pi","workspace":false}}}}                                                                                                           TRACE Started scanning user extensions                                                                                  
[2021-06-25T03:52:37.704Z] trace got message {"message":{"type":"options","id":"PVkS1Xb5TRcuzXLB4FI1QbvO","options":{"workbenchWebConfiguration":{"folderUri":{"$mid":1,"path":"/media/pi","scheme":"vscode-remote","authority":"vbc.shmoo.dev"},"remoteAuthority":"vbc.shmoo.dev","logLevel":0,"workspaceProvider":{"payload":[["userDataPath","/home/pi/.local/share/code-server"],["enableProposedApi","[]"]]}},"remoteUserDataUri":{"$mid":1,"path":"/home/pi/.local/share/code-server","scheme":"vscode-remote","authority":"vbc.shmoo.dev"},"productConfiguration":{"nameShort":"code-server","nameLong":"code-server","applicationName":"code-oss","dataFolderName":".vscode-oss","win32MutexName":"vscodeoss","licenseName":"MIT","licenseUrl":"https://github.com/microsoft/vscode/blob/main/LICENSE.txt","win32DirName":"Microsoft Code OSS","win32NameVersion":"Microsoft Code OSS","win32RegValueName":"CodeOSS","win32AppId":"{{E34003BB-9E10-4501-8C11-BE3FAA83F23F}","win32x64AppId":"{{D77B7E06-80BA-4137-BCF4-654B95CCEBC5}","win32arm64AppId":"{{D1ACE434-89C5-48D1-88D3-E2991DF85475}","win32UserAppId":"{{C6065F05-9603-4FC4-8101-B9781A25D88E}","win32x64UserAppId":"{{CC6B787D-37A0-49E8-AE24-8559A032BE0C}","win32arm64UserAppId":"{{3AEBF0C8-F733-4AD4-BADE-FDB816D53D7B}","win32AppUserModelId":"Microsoft.CodeOSS","win32ShellNameShort":"C&ode - OSS","darwinBundleIdentifier":"com.visualstudio.code.oss","linuxIconName":"com.visualstudio.code.oss","licenseFileName":"LICENSE.txt","reportIssueUrl":"https://github.com/cdr/code-server/issues/new","urlProtocol":"code-oss","extensionAllowedProposedApi":["ms-vscode.vscode-js-profile-flame","ms-vscode.vscode-js-profile-table","ms-vscode.github-browser","ms-vscode.github-richnav","ms-vscode.remotehub","ms-vscode.remotehub-insiders"],"builtInExtensions":[{"name":"ms-vscode.node-debug","version":"1.44.27","repo":"https://github.com/microsoft/vscode-node-debug","metadata":{"id":"b6ded8fb-a0a0-4c1c-acbd-ab2a3bc995a6","publisherId":{"publisherId":"5f5636e7-69ed-4afe-b5d6-8d231fb3d3ee","publisherName":"ms-vscode","displayName":"Microsoft","flags":"verified"},"publisherDisplayName":"Microsoft"}},{"name":"ms-vscode.node-debug2","version":"1.42.6","repo":"https://github.com/microsoft/vscode-node-debug2","metadata":{"id":"36d19e17-7569-4841-a001-947eb18602b2","publisherId":{"publisherId":"5f5636e7-69ed-4afe-b5d6-8d231fb3d3ee","publisherName":"ms-vscode","displayName":"Microsoft","flags":"verified"},"publisherDisplayName":"Microsoft"}},{"name":"ms-vscode.references-view","version":"0.0.80","repo":"https://github.com/microsoft/vscode-references-view","metadata":{"id":"dc489f46-520d-4556-ae85-1f9eab3c412d","publisherId":{"publisherId":"5f5636e7-69ed-4afe-b5d6-8d231fb3d3ee","publisherName":"ms-vscode","displayName":"Microsoft","flags":"verified"},"publisherDisplayName":"Microsoft"}},{"name":"ms-vscode.js-debug-companion","version":"1.0.13","repo":"https://github.com/microsoft/vscode-js-debug-companion","metadata":{"id":"99cb0b7f-7354-4278-b8da-6cc79972169d","publisherId":{"publisherId":"5f5636e7-69ed-4afe-b5d6-8d231fb3d3ee","publisherName":"ms-vscode","displayName":"Microsoft","flags":"verified"},"publisherDisplayName":"Microsoft"}},{"name":"ms-vscode.js-debug","version":"1.56.2","repo":"https://github.com/microsoft/vscode-js-debug","metadata":{"id":"25629058-ddac-4e17-abba-74678e126c5d","publisherId":{"publisherId":"5f5636e7-69ed-4afe-b5d6-8d231fb3d3ee","publisherName":"ms-vscode","displayName":"Microsoft","flags":"verified"},"publisherDisplayName":"Microsoft"}},{"name":"ms-vscode.vscode-js-profile-table","version":"0.0.18","repo":"https://github.com/microsoft/vscode-js-profile-visualizer","metadata":{"id":"7e52b41b-71ad-457b-ab7e-0620f1fc4feb","publisherId":{"publisherId":"5f5636e7-69ed-4afe-b5d6-8d231fb3d3ee","publisherName":"ms-vscode","displayName":"Microsoft","flags":"verified"},"publisherDisplayName":"Microsoft"}},{"name":"ms-vscode.remotehub","version":"0.5.3","repo":"https://github.com/microsoft/vscode-remotehub","metadata":{"id":"ed0ffe1d-36f0-49a0-bafe-da68355eb33a","publisherId":{"publisherId":"5f5636e7-69ed-4afe-b5d6-8d231fb3d3ee","publisherName":"ms-vscode","displayName":"Microsoft","flags":"verified"},"publisherDisplayName":"Microsoft"}}],"webBuiltInExtensions":[{"name":"ms-vscode.remotehub","version":"0.5.3","repo":"https://github.com/microsoft/vscode-remotehub","metadata":{"id":"ed0ffe1d-36f0-49a0-bafe-da68355eb33a","publisherId":{"publisherId":"5f5636e7-69ed-4afe-b5d6-8d231fb3d3ee","publisherName":"ms-vscode","displayName":"Microsoft","flags":"verified"},"publisherDisplayName":"Microsoft"}}],"//":"https://github.com/VSCodium/vscodium/pull/155/files","documentationUrl":"https://go.microsoft.com/fwlink/?LinkID=533484#vscode","keyboardShortcutsUrlMac":"https://go.microsoft.com/fwlink/?linkid=832143","keyboardShortcutsUrlLinux":"https://go.microsoft.com/fwlink/?linkid=832144","keyboardShortcutsUrlWin":"https://go.microsoft.com/fwlink/?linkid=832145","introductoryVideosUrl":"https://go.microsoft.com/fwlink/?linkid=832146","tipsAndTricksUrl":"https://go.microsoft.com/fwlink/?linkid=852118","newsletterSignupUrl":"https://www.research.net/r/vsc-newsletter","commit":"387b12ef4ca404ffd39d84834e1f0776e9e3c005","date":"2021-05-21T23:13:00Z","version":"1.56.1","extensionsGallery":{"serviceUrl":"https://extensions.coder.com/api","itemUrl":"","controlUrl":"","recommendationsUrl":""}},"nlsConfiguration":{"locale":"en","availableLanguages":{}},"commit":"387b12ef4ca404ffd39d84834e1f0776e9e3c005"}}}                                     TRACE Scanned user extensions: 19                                                                                       
[2021-06-25T03:52:39.838Z] debug vscode got message from code-server {"type":"socket"}                                  
[2021-06-25T03:52:39.839Z] trace vscode code-server message content {"message":{"type":"socket","query":{"type":"Management","reconnectionToken":"REDACTED","reconnection":"false","skipWebSocketFrames":"false"},"permessageDeflate":false}}                                                                                               
[2021-06-25T03:52:39.843Z] debug protocol Initiating handshake... {"token":"REDACTED"}      
[2021-06-25T03:52:39.903Z] trace protocol Got message {"message":"{\"type\":\"auth\",\"auth\":\"00000000000000000000\"}","token":"REDACTED"}                                                                        
[2021-06-25T03:52:39.906Z] trace protocol Got message {"message":"{\"type\":\"connectionType\",\"commit\":\"387b12ef4ca404ffd39d84834e1f0776e9e3c005\",\"signedData\":\"\",\"desiredConnectionType\":1}","token":"REDACTED"}                                                                                                                
[2021-06-25T03:52:39.907Z] debug protocol Handshake completed {"token":"REDACTED"}          
[2021-06-25T03:52:39.908Z] debug management Connecting... {"token":"REDACTED"}              
[2021-06-25T03:52:39.909Z] debug vscode 1 active management connection(s)                                               TRACE Started scanning system extensions                                                                                TRACE Started scanning user extensions                                                                                  TRACE Started scanning system extensions                                                                                TRACE Started scanning user extensions                                                                                  TRACE Started scanning system extensions                                                                                TRACE Started scanning user extensions                                                                                  
[2021-06-25T03:52:40.861Z] trace TerminalProviderChannel:call {"command":"$getTerminalLayoutInfo","args":{"workspaceId":"4f1a9245"}}                                                                                                            
[2021-06-25T03:52:40.865Z] trace TerminalProviderChannel:listen {"event":"$onProcessDataEvent"}                         
[2021-06-25T03:52:40.865Z] trace TerminalProviderChannel:listen {"event":"$onProcessExitEvent"}                         
[2021-06-25T03:52:40.867Z] trace TerminalProviderChannel:listen {"event":"$onProcessReadyEvent"}                        
[2021-06-25T03:52:40.868Z] trace TerminalProviderChannel:listen {"event":"$onProcessTitleChangedEvent"}                 
[2021-06-25T03:52:40.869Z] trace TerminalProviderChannel:listen {"event":"$onProcessShellTypeChangedEvent"}             
[2021-06-25T03:52:40.870Z] trace TerminalProviderChannel:listen {"event":"$onProcessOverrideDimensionsEvent"}           
[2021-06-25T03:52:40.870Z] trace TerminalProviderChannel:listen {"event":"$onProcessResolvedShellLaunchConfigEvent"}    
[2021-06-25T03:52:40.871Z] trace TerminalProviderChannel:listen {"event":"$onProcessReplayEvent"}                       
[2021-06-25T03:52:40.873Z] trace TerminalProviderChannel:listen {"event":"$onProcessOrphanQuestion"}                    
[2021-06-25T03:52:40.874Z] trace TerminalProviderChannel:listen {"event":"$onExecuteCommand"}                           
[2021-06-25T03:52:40.874Z] trace TerminalProviderChannel:listen {"event":"$onPtyHostExitEvent"}                         
[2021-06-25T03:52:40.875Z] trace TerminalProviderChannel:listen {"event":"$onPtyHostStartEvent"}                        
[2021-06-25T03:52:40.876Z] trace TerminalProviderChannel:listen {"event":"$onPtyHostUnresponsiveEvent"}                 
[2021-06-25T03:52:40.876Z] trace TerminalProviderChannel:listen {"event":"$onPtyHostResponsiveEvent"}                   
[2021-06-25T03:52:40.891Z] debug got latest version {"latest":"3.10.2"}                                                 
[2021-06-25T03:52:40.892Z] debug comparing versions {"current":"3.10.2","latest":"3.10.2"}                              
[2021-06-25T03:52:41.072Z] debug vscode got message from code-server {"type":"socket"}                                  
[2021-06-25T03:52:41.073Z] trace vscode code-server message content {"message":{"type":"socket","query":{"type":"ExtensionHost","reconnectionToken":"8168ea39-f1e9-40ca-b714-a0ea4c19f4b5","reconnection":"false","skipWebSocketFrames":"false"},"permessageDeflate":false}}                                                                                            
[2021-06-25T03:52:41.074Z] debug protocol Initiating handshake... {"token":"8168ea39-f1e9-40ca-b714-a0ea4c19f4b5"}      
[2021-06-25T03:52:41.158Z] trace TerminalProviderChannel:call {"command":"$reduceConnectionGraceTime"}                  
[2021-06-25T03:52:41.226Z] trace protocol Got message {"message":"{\"type\":\"connectionType\",\"commit\":\"387b12ef4ca404ffd39d84834e1f0776e9e3c005\",\"signedData\":\"\",\"desiredConnectionType\":2,\"args\":{\"language\":\"en\",\"break\":false,\"port\":null}}","token":"8168ea39-f1e9-40ca-b714-a0ea4c19f4b5"}                                                   
[2021-06-25T03:52:41.228Z] debug protocol Handshake completed {"token":"8168ea39-f1e9-40ca-b714-a0ea4c19f4b5"}          
[2021-06-25T03:52:41.229Z] debug exthost Connecting... {"token":"8168ea39-f1e9-40ca-b714-a0ea4c19f4b5"}                 
[2021-06-25T03:52:41.233Z] debug exthost Getting NLS configuration... {"token":"8168ea39-f1e9-40ca-b714-a0ea4c19f4b5"}  
[2021-06-25T03:52:41.234Z] debug vscode 1 active exthost connection(s)                                                  
[2021-06-25T03:52:41.235Z] debug exthost Spawning extension host... {"token":"8168ea39-f1e9-40ca-b714-a0ea4c19f4b5"}    
[2021-06-25T03:52:41.266Z] debug exthost Waiting for handshake... {"token":"8168ea39-f1e9-40ca-b714-a0ea4c19f4b5"}      
[2021-06-25T03:52:41.316Z] trace TerminalProviderChannel:call {"command":"$getShellEnvironment"}                        
[2021-06-25T03:52:41.549Z] trace TerminalProviderChannel:call {"command":"$getShellEnvironment"}                        
[2021-06-25T03:52:41.687Z] trace TerminalProviderChannel:call {"command":"$createProcess","args":{"configuration":{"terminal.integrated.automationShell.windows":{},"terminal.integrated.automationShell.osx":{},"terminal.integrated.automationShell.linux":{},"terminal.integrated.shell.windows":{},"terminal.integrated.shell.osx":{},"terminal.integrated.shell.linux":"/bin/bash","terminal.integrated.shellArgs.windows":[],"terminal.integrated.shellArgs.osx":["-l"],"terminal.integrated.shellArgs.linux":[],"terminal.integrated.env.windows":{},"terminal.integrated.env.osx":{},"terminal.integrated.env.linux":{},"terminal.integrated.inheritEnv":false,"terminal.integrated.cwd":"","terminal.integrated.detectLocale":"auto"},"resolvedVariables":{},"envVariableCollections":[["vscode.git",[["VSCODE_GIT_IPC_HANDLE",{"value":"/run/user/1000/vscode-git-03938d6028.sock","type":1}],["GIT_ASKPASS",{"value":"/usr/lib/code-server/lib/vscode/extensions/git/dist/askpass.sh","type":1}],["VSCODE_GIT_ASKPASS_NODE",{"value":"/usr/lib/code-server/lib/node","type":1}],["VSCODE_GIT_ASKPASS_MAIN",{"value":"/usr/lib/code-server/lib/vscode/extensions/git/dist/askpass-main.js","type":1}]]]],"shellLaunchConfig":{"executable":"/bin/bash","args":[]},"workspaceId":"4f1a9245","workspaceName":"pi","workspaceFolders":[{"uri":{"$mid":1,"external":"vscode-remote://vbc.shmoo.dev/media/pi","path":"/media/pi","scheme":"vscode-remote","authority":"vbc.shmoo.dev"},"name":"pi","index":0}],"activeWorkspaceFolder":{"uri":{"$mid":1,"external":"vscode-remote://vbc.shmoo.dev/media/pi","path":"/media/pi","scheme":"vscode-remote","authority":"vbc.shmoo.dev"},"name":"pi","index":0},"shouldPersistTerminal":true,"cols":102,"rows":14}}                                                                                                    
[2021-06-25T03:52:41.917Z] trace TerminalProviderChannel:call {"command":"$setTerminalLayoutInfo","args":{"workspaceId":"4f1a9245","tabs":[{"isActive":true,"terminals":[]}]}}                                                                  
[2021-06-25T03:52:42.507Z] trace TerminalProviderChannel:call {"command":"$start","args":[1]}                           
[2021-06-25T03:52:42.742Z] trace TerminalProviderChannel:call {"command":"$resize","args":[1,102,14]}                   
[2021-06-25T03:52:43.234Z] trace TerminalProviderChannel:call {"command":"$setTerminalLayoutInfo","args":{"workspaceId":"4f1a9245","tabs":[{"isActive":true,"activePersistentProcessId":1,"terminals":[{"relativeSize":1,"terminal":1}]}]}}     
[2021-06-25T03:52:43.237Z] debug exthost Handshake completed {"token":"8168ea39-f1e9-40ca-b714-a0ea4c19f4b5"}           
[2021-06-25T03:52:43.238Z] debug exthost Sending socket {"token":"8168ea39-f1e9-40ca-b714-a0ea4c19f4b5"}                
[2021-06-25T03:52:43.345Z] debug management Disconnected {"token":"REDACTED"}               
[2021-06-25T03:52:43.547Z] debug vscode got message from code-server {"type":"socket"}                                  
[2021-06-25T03:52:43.548Z] trace vscode code-server message content {"message":{"type":"socket","query":{"type":"Management","reconnectionToken":"REDACTED","reconnection":"true","skipWebSocketFrames":"false"},"permessageDeflate":false}}                                                                                                
[2021-06-25T03:52:43.550Z] debug protocol Initiating handshake... {"token":"REDACTED"}      
[2021-06-25T03:52:43.625Z] trace protocol Got message {"message":"{\"type\":\"auth\",\"auth\":\"00000000000000000000\"}","token":"REDACTED"}                                                                        
[2021-06-25T03:52:43.630Z] trace protocol Got message {"message":"{\"type\":\"connectionType\",\"commit\":\"387b12ef4ca404ffd39d84834e1f0776e9e3c005\",\"signedData\":\"\",\"desiredConnectionType\":1}","token":"REDACTED"}                                                                                                                
[2021-06-25T03:52:43.631Z] debug protocol Handshake completed {"token":"REDACTED"}          
[2021-06-25T03:52:43.632Z] debug management Reconnecting... {"token":"REDACTED"}            
[2021-06-25T03:52:43.708Z] warn  vscode Uncaught error {"error":"read ECONNRESET"}                                      
[2021-06-25T03:52:43.711Z] debug management Disconnected {"token":"REDACTED"}               
[2021-06-25T03:52:43.947Z] debug vscode got message from code-server {"type":"socket"}                                  
[2021-06-25T03:52:43.948Z] trace vscode code-server message content {"message":{"type":"socket","query":{"type":"Management","reconnectionToken":"REDACTED","reconnection":"true","skipWebSocketFrames":"false"},"permessageDeflate":false}}                                                                                                
[2021-06-25T03:52:43.949Z] debug protocol Initiating handshake... {"token":"REDACTED"}      
[2021-06-25T03:52:44.008Z] trace protocol Got message {"message":"{\"type\":\"auth\",\"auth\":\"00000000000000000000\"}","token":"REDACTED"}                                                                        
[2021-06-25T03:52:44.013Z] trace protocol Got message {"message":"{\"type\":\"connectionType\",\"commit\":\"387b12ef4ca404ffd39d84834e1f0776e9e3c005\",\"signedData\":\"\",\"desiredConnectionType\":1}","token":"REDACTED"}                                                                                                                
[2021-06-25T03:52:44.014Z] debug protocol Handshake completed {"token":"REDACTED"}          
[2021-06-25T03:52:44.015Z] debug management Reconnecting... {"token":"REDACTED"}            
[2021-06-25T03:52:44.120Z] warn  vscode Uncaught error {"error":"write ECONNRESET"}                                     
[2021-06-25T03:52:44.123Z] warn  vscode Uncaught error {"error":"write ECONNRESET"}                                     
[2021-06-25T03:52:44.124Z] debug management Disconnected {"token":"REDACTED"}               
[2021-06-25T03:52:44.358Z] debug vscode got message from code-server {"type":"socket"}                                  
[2021-06-25T03:52:44.359Z] trace vscode code-server message content {"message":{"type":"socket","query":{"type":"Management","reconnectionToken":"REDACTED","reconnection":"true","skipWebSocketFrames":"false"},"permessageDeflate":false}}                                                                                                
[2021-06-25T03:52:44.360Z] debug protocol Initiating handshake... {"token":"REDACTED"}      
[2021-06-25T03:52:44.419Z] trace protocol Got message {"message":"{\"type\":\"auth\",\"auth\":\"00000000000000000000\"}","token":"REDACTED"}                                                                        
[2021-06-25T03:52:44.426Z] trace protocol Got message {"message":"{\"type\":\"connectionType\",\"commit\":\"387b12ef4ca404ffd39d84834e1f0776e9e3c005\",\"signedData\":\"\",\"desiredConnectionType\":1}","token":"REDACTED"}                                                                                                                
[2021-06-25T03:52:44.427Z] debug protocol Handshake completed {"token":"REDACTED"}          
[2021-06-25T03:52:44.427Z] debug management Reconnecting... {"token":"REDACTED"}            
[2021-06-25T03:52:44.503Z] warn  vscode Uncaught error {"error":"write ECONNRESET"}                                     
[2021-06-25T03:52:44.504Z] warn  vscode Uncaught error {"error":"write ECONNRESET"}                                     
[2021-06-25T03:52:44.505Z] debug management Disconnected {"token":"REDACTED"}               
[2021-06-25T03:52:44.717Z] debug vscode got message from code-server {"type":"socket"}                                  
[2021-06-25T03:52:44.718Z] trace vscode code-server message content {"message":{"type":"socket","query":{"type":"Management","reconnectionToken":"REDACTED","reconnection":"true","skipWebSocketFrames":"false"},"permessageDeflate":false}}                                                                                                
[2021-06-25T03:52:44.719Z] debug protocol Initiating handshake... {"token":"REDACTED"}      
[2021-06-25T03:52:44.771Z] trace protocol Got message {"message":"{\"type\":\"auth\",\"auth\":\"00000000000000000000\"}","token":"REDACTED"}                                                                        
[2021-06-25T03:52:44.775Z] trace protocol Got message {"message":"{\"type\":\"connectionType\",\"commit\":\"387b12ef4ca404ffd39d84834e1f0776e9e3c005\",\"signedData\":\"\",\"desiredConnectionType\":1}","token":"REDACTED"}                                                                                                                
[2021-06-25T03:52:44.776Z] debug protocol Handshake completed {"token":"REDACTED"}          
[2021-06-25T03:52:44.777Z] debug management Reconnecting... {"token":"REDACTED"}            
[2021-06-25T03:52:44.847Z] debug management Disconnected {"token":"REDACTED"}               
[2021-06-25T03:52:45.101Z] debug vscode got message from code-server {"type":"socket"}                                  
[2021-06-25T03:52:45.102Z] trace vscode code-server message content {"message":{"type":"socket","query":{"type":"Management","reconnectionToken":"REDACTED","reconnection":"true","skipWebSocketFrames":"false"},"permessageDeflate":false}}                                                                                                
[2021-06-25T03:52:45.102Z] debug protocol Initiating handshake... {"token":"REDACTED"}      
[2021-06-25T03:52:45.161Z] trace protocol Got message {"message":"{\"type\":\"auth\",\"auth\":\"00000000000000000000\"}","token":"REDACTED"}                                                                        
[2021-06-25T03:52:45.166Z] trace protocol Got message {"message":"{\"type\":\"connectionType\",\"commit\":\"387b12ef4ca404ffd39d84834e1f0776e9e3c005\",\"signedData\":\"\",\"desiredConnectionType\":1}","token":"REDACTED"}                                                                                                                
[2021-06-25T03:52:45.167Z] debug protocol Handshake completed {"token":"REDACTED"}          
[2021-06-25T03:52:45.168Z] debug management Reconnecting... {"token":"REDACTED"}            
[2021-06-25T03:52:45.244Z] warn  vscode Uncaught error {"error":"write ECONNRESET"}                                     
[2021-06-25T03:52:45.245Z] warn  vscode Uncaught error {"error":"write ECONNRESET"}                                     
[2021-06-25T03:52:45.246Z] debug management Disconnected {"token":"REDACTED"}               
[2021-06-25T03:52:45.503Z] debug vscode got message from code-server {"type":"socket"}                                  
[2021-06-25T03:52:45.504Z] trace vscode code-server message content {"message":{"type":"socket","query":{"type":"Management","reconnectionToken":"REDACTED","reconnection":"true","skipWebSocketFrames":"false"},"permessageDeflate":false}}                                                                                                
[2021-06-25T03:52:45.505Z] debug protocol Initiating handshake... {"token":"REDACTED"}      
[2021-06-25T03:52:45.557Z] trace protocol Got message {"message":"{\"type\":\"auth\",\"auth\":\"00000000000000000000\"}","token":"REDACTED"}                                                                        
[2021-06-25T03:52:45.568Z] trace protocol Got message {"message":"{\"type\":\"connectionType\",\"commit\":\"387b12ef4ca404ffd39d84834e1f0776e9e3c005\",\"signedData\":\"\",\"desiredConnectionType\":1}","token":"REDACTED"}                                                                                                                
[2021-06-25T03:52:45.569Z] debug protocol Handshake completed {"token":"REDACTED"}          
[2021-06-25T03:52:45.569Z] debug management Reconnecting... {"token":"REDACTED"}            
[2021-06-25T03:52:45.649Z] warn  vscode Uncaught error {"error":"read ECONNRESET"}                                      
[2021-06-25T03:52:45.650Z] debug management Disconnected {"token":"REDACTED"}               
[2021-06-25T03:52:45.824Z] debug vscode got message from code-server {"type":"socket"}                                  
[2021-06-25T03:52:45.824Z] trace vscode code-server message content {"message":{"type":"socket","query":{"type":"Management","reconnectionToken":"REDACTED","reconnection":"true","skipWebSocketFrames":"false"},"permessageDeflate":false}}                                                                                                
[2021-06-25T03:52:45.825Z] debug protocol Initiating handshake... {"token":"REDACTED"}      
[2021-06-25T03:52:45.887Z] trace protocol Got message {"message":"{\"type\":\"auth\",\"auth\":\"00000000000000000000\"}","token":"REDACTED"}                                                                        
[2021-06-25T03:52:45.888Z] trace protocol Got message {"message":"{\"type\":\"connectionType\",\"commit\":\"387b12ef4ca404ffd39d84834e1f0776e9e3c005\",\"signedData\":\"\",\"desiredConnectionType\":1}","token":"REDACTED"}                                                                                                                
[2021-06-25T03:52:45.889Z] debug protocol Handshake completed {"token":"REDACTED"}          
[2021-06-25T03:52:45.890Z] debug management Reconnecting... {"token":"REDACTED"}            
[2021-06-25T03:52:45.962Z] warn  vscode Uncaught error {"error":"write ECONNRESET"}                                     
[2021-06-25T03:52:45.963Z] warn  vscode Uncaught error {"error":"write ECONNRESET"}                                     
[2021-06-25T03:52:45.964Z] debug management Disconnected {"token":"REDACTED"}               
[2021-06-25T03:52:46.218Z] debug vscode got message from code-server {"type":"socket"}                                  
[2021-06-25T03:52:46.219Z] trace vscode code-server message content {"message":{"type":"socket","query":{"type":"Management","reconnectionToken":"REDACTED","reconnection":"true","skipWebSocketFrames":"false"},"permessageDeflate":false}}                                                                                                
[2021-06-25T03:52:46.220Z] debug protocol Initiating handshake... {"token":"REDACTED"}      
[2021-06-25T03:52:46.279Z] trace protocol Got message {"message":"{\"type\":\"auth\",\"auth\":\"00000000000000000000\"}","token":"REDACTED"}                                                                        
[2021-06-25T03:52:46.282Z] trace protocol Got message {"message":"{\"type\":\"connectionType\",\"commit\":\"387b12ef4ca404ffd39d84834e1f0776e9e3c005\",\"signedData\":\"\",\"desiredConnectionType\":1}","token":"REDACTED"}                                                                                                                
[2021-06-25T03:52:46.283Z] debug protocol Handshake completed {"token":"REDACTED"}          
[2021-06-25T03:52:46.284Z] debug management Reconnecting... {"token":"REDACTED"}            
[2021-06-25T03:52:46.356Z] warn  vscode Uncaught error {"error":"write ECONNRESET"}                                     
[2021-06-25T03:52:46.357Z] warn  vscode Uncaught error {"error":"write ECONNRESET"}                                     
[2021-06-25T03:52:46.358Z] debug management Disconnected {"token":"REDACTED"}               
[2021-06-25T03:52:46.592Z] debug vscode got message from code-server {"type":"socket"}                                  
[2021-06-25T03:52:46.593Z] trace vscode code-server message content {"message":{"type":"socket","query":{"type":"Management","reconnectionToken":"REDACTED","reconnection":"true","skipWebSocketFrames":"false"},"permessageDeflate":false}}                                                                                                
[2021-06-25T03:52:46.594Z] debug protocol Initiating handshake... {"token":"REDACTED"}      
[2021-06-25T03:52:46.654Z] trace protocol Got message {"message":"{\"type\":\"auth\",\"auth\":\"00000000000000000000\"}","token":"REDACTED"}                                                                        
[2021-06-25T03:52:46.655Z] trace protocol Got message {"message":"{\"type\":\"connectionType\",\"commit\":\"387b12ef4ca404ffd39d84834e1f0776e9e3c005\",\"signedData\":\"\",\"desiredConnectionType\":1}","token":"REDACTED"}                                                                                                                
[2021-06-25T03:52:46.655Z] debug protocol Handshake completed {"token":"REDACTED"}          
[2021-06-25T03:52:46.656Z] debug management Reconnecting... {"token":"REDACTED"}            
[2021-06-25T03:52:46.737Z] warn  vscode Uncaught error {"error":"write ECONNRESET"}                                     
[2021-06-25T03:52:46.739Z] warn  vscode Uncaught error {"error":"write ECONNRESET"}                                     
[2021-06-25T03:52:46.740Z] debug management Disconnected {"token":"REDACTED"}               
[2021-06-25T03:52:46.932Z] debug vscode got message from code-server {"type":"socket"}                                  
[2021-06-25T03:52:46.932Z] trace vscode code-server message content {"message":{"type":"socket","query":{"type":"Management","reconnectionToken":"REDACTED","reconnection":"true","skipWebSocketFrames":"false"},"permessageDeflate":false}}                                                                                                
[2021-06-25T03:52:46.933Z] debug protocol Initiating handshake... {"token":"REDACTED"}      
[2021-06-25T03:52:46.993Z] trace protocol Got message {"message":"{\"type\":\"auth\",\"auth\":\"00000000000000000000\"}","token":"REDACTED"}                                                                        
[2021-06-25T03:52:46.999Z] trace protocol Got message {"message":"{\"type\":\"connectionType\",\"commit\":\"387b12ef4ca404ffd39d84834e1f0776e9e3c005\",\"signedData\":\"\",\"desiredConnectionType\":1}","token":"REDACTED"}                                                                                                                ^C
[2021-06-25T03:52:47.004Z] debug child:32529 disposing {"code":"SIGINT"}

**Comment by RWejlgaard** [comment]
I’ve been lurking on this issue, I have the same issue. I believe it could be related to the Private Relay settings in iOS 15 where apple is protecting users using Apple VPNs, I’ve tried everything to turn it off but with no luck.
It’s possible that code-server doesn’t handle every request coming from different IPs very well.

**Comment by RWejlgaard** [comment]
Guys I think I found a solution, in iOS settings go to `Safari -> Advanced -> Experimental Features` and disable `NSURLSession WebSocket`.

This worked on my iPhone running iOS 15 beta 2.

I don’t know enough about how this works but it sounds like it’s changing the way safari handles websockets establishing.

**Comment by tommycusick** [comment]
Confirmed that the workaround suggested by @RWejlgaard works on iOS 15 beta 2.  Filed Safari beta feedback with Apple (FB9240512) and referenced this bug report.

**Comment by ericbuess** [comment]
@RWejlgaard’s workaround also works for me also — iOS 15 beta 2

**Comment by Cattttttttt** [comment]
Thanks guys I tried your solutions and I can use code-server in Safari now.

But serveditor still got the issue.

**Comment by reticulatingsplines** [comment]
This workaround also worked for me on 14.6. However, the terminal window is still blank (no prompt) when I access from Safari from either my iPad 14.6 (M1 12.9") or from a M1 Mac mini running macos Big Sur. Maybe unrelated issue but I'm at wits' end troubleshooting it.

**Comment by jsjoeio** [comment]
@reticulatingsplines do you mind opening a separate issue and filing a bug report?

**Comment by AndresRQ27** [comment]
> This workaround also worked for me on 14.6. However, the terminal window is still blank (no prompt) when I access from Safari from either my iPad 14.6 (M1 12.9") or from a M1 Mac mini running macos Big Sur. Maybe unrelated issue but I'm at wits' end troubleshooting it.

If it's the same problem I had with the terminal window, you can fix it by changing the setting: `Settings --> Terminal --> Integrated: Gpu Acceleration --> Turn off`

**Comment by ericbuess** [comment]
> > This workaround also worked for me on 14.6. However, the terminal window is still blank (no prompt) when I access from Safari from either my iPad 14.6 (M1 12.9") or from a M1 Mac mini running macos Big Sur. Maybe unrelated issue but I'm at wits' end troubleshooting it.
> 
> 
> 
> If it's the same problem I had with the terminal window, you can fix it by changing the setting: `Settings --> Terminal --> Integrated: Gpu Acceleration --> Turn off`

That also worked for me!
